nvflare/app_common/executors/task_script_runner.py

```diff
@@ -14,75 +14,101 @@
+        self.script_path = script_path
+        self.script_full_path = self.get_script_full_path(self.site_name, self.script_path)
 
     def run(self):
         """Call the task_fn with any required arguments."""
-        self.logger.info(f"\n start task run() with {self.script_path}")
+        self.logger.info(f"\n start task run() with full path: {self.script_full_path}")
         try:
-            import runpy
-
             curr_argv = sys.argv
-            builtins.print = log_print
+            builtins.print = log_print if self.redirect_print_to_log else print_fn
             sys.argv = self.get_sys_argv()
-            runpy.run_path(self.script_path, run_name="__main__")
+            runpy.run_path(self.script_full_path, run_name="__main__")
             sys.argv = curr_argv
-
+        except ImportError as ie:
+            msg = "attempted relative import with no known parent package"
+            if ie.msg == msg:
+                xs = [p for p in sys.path if self.script_full_path.startswith(p)]
+                import_base_path = max(xs, key=len)
+                raise ImportError(
+                    f"{ie.msg}, the relative import is not support. python import is based off the sys.path: {import_base_path}"
+                )
+            else:
+                raise ie
         except Exception as e:
             msg = traceback.format_exc()
             self.logger.error(msg)
-            if self.client_api:
-                self.client_api.exec_queue.ask_abort(msg)
+            self.logger.error("fire abort event")
+            self.event_manager.fire_event(TOPIC_ABORT, f"'{self.script_full_path}' is aborted, {msg}")
             raise e
         finally:
             builtins.print = print_fn
 
     def get_sys_argv(self):
         args_list = [] if not self.script_args else self.script_args.split()
-        return [self.script_path] + args_list
+        return [self.script_full_path] + args_list
 
-    def get_script_full_path(self, script_path) -> str:
+    def get_script_full_path(self, site_name, script_path) -> str:
         target_file = None
         script_filename = os.path.basename(script_path)
         script_dirs = os.path.dirname(script_path)
 
+        if os.path.isabs(script_path):
+            if not os.path.isfile(script_path):
+                raise ValueError(f"script_path='{script_path}' not found")
+            return script_path
+
         for r, dirs, files in os.walk(os.getcwd()):
             for f in files:
                 absolute_path = os.path.join(r, f)
                 if absolute_path.endswith(script_path):
                     parent_dir = absolute_path[: absolute_path.find(script_path)].rstrip(os.sep)
                     if os.path.isdir(parent_dir):
-                        target_file = absolute_path
-                        break
+                        path_components = parent_dir.split(os.path.sep)
+                        if site_name in path_components:
```

Comment:
So our site-name can not contain path separators, do we specify our limitations somewhere in our documentation @SYangster